src/main/java/io/appform/dropwizard/actors/base/UnmanagedPublisher.java

```diff
@@ -106,6 +113,79 @@ public final long pendingSidelineMessagesCount() {
         return Long.MAX_VALUE;
     }
 
+    /**
+     * Note: Timeout is in MilliSeconds, Function take a list of message as input and return not ack msg as output
+     * @param messages : Messages to be published
+     * @param properties
+     * @param timeout : in MS timeout for waiting on countDownLatch
+     * @param unit : timeout unit
+     * @return : List of message nacked
+     * @throws Exception
+     */
+    public final List<Message> publishWithConfirmListener(List<Message> messages, AMQP.BasicProperties properties,
+            long timeout, @NotNull TimeUnit unit) throws Exception {
+        publishChannel.confirmSelect();
+        ConcurrentNavigableMap<Long, Message> outstandingConfirms = new ConcurrentSkipListMap<>();
+        final CountDownLatch publishAckLatch = new CountDownLatch(messages.size());
+        publishChannel.addConfirmListener((sequenceNumber, multiple) -> {
+            if (multiple) {
```

Comment:
Make the two lambdas into separate methods, they are pretty big, give a good name to indicate what they are doing

Reply:
done